File: mobile/modules/quantum-secure/src/QuantumSecureModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class QuantumSecureModule extends NativeModule<{}> {
  setValueAsync(key: string, value: string): Promise<void>;
  getValueAsync(key: string): Promise<string | null>;
}

export default requireNativeModule<QuantumSecureModule>('QuantumSecure');

File: mobile/modules/quantum-secure/android/src/main/java/expo/modules/quantumsecure/QuantumSecureModule.kt
package expo.modules.quantumsecure

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class QuantumSecureModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("QuantumSecure")

    AsyncFunction("setValueAsync") { value: String ->
    }
  }
}
